profiling/space-time-stack/kp_space_time_stack.cpp

```diff
@@ -336,9 +336,11 @@ struct StackNode {
   void print_recursive(std::ostream& os, std::string my_indent,
                        std::string const& child_indent,
                        double tree_time) const {
-    auto percent = (total_runtime / tree_time) * 100.0;
+    const double comm_size = total_runtime / avg_runtime;
+    auto threshold_percent = ((max_runtime * comm_size) / tree_time) * 100.0;
+    auto percent           = (total_runtime / tree_time) * 100.0;
```

Comment:
Can you elaborate on why this is sensible? I understand the previous code but now we essentially do
```
((max_runtime / avg_runtime) * percent  < outout_threshold
```
In other words, why does multiplying by `max_runtime / avg_runtime` do what you want? Wouldn't you want to do something like
```
if (percent > first_threshold || (high_imbalance && percent > second_threshold))
  report
```
? It's not quite clear to me that this second threshold should be the same as the orignal one divided by the imbalance. Maybe it's sensible to report all kernels that have an imbalance greater than a certain factor?

Reply:
The thinking is to identify any kernels/regions that would be above the 0.1% threshold if we only looked at the runtimes on that rank. Assuming that total_runtime is evenly distributed across ranks (which I think is basically always true), then `rank_wall_time = tree_time / comm_size` and therefore `max_runtime / rank_wall_time == (max_runtime * comm_size) / tree_time`.

Reply:
OK. You basically want to use `max_runtime` instead of `avg_runtime` for comparing with the threshold value. I think that's sensible. Not sure if we need a configuration option.
Note that there also is `print_json_recursive` that needs consistent logic.